pkg/gridtypes/zos/gw_fqdn.go

```diff
@@ -17,28 +19,46 @@ var (
 
 type Backend string
 
-// check if valid http://ip:port, http://ip or ip:port
-func (b Backend) Valid(tls bool) error {
-	u, err := url.Parse(string(b))
-	if err != nil {
-		return errors.Wrap(err, "failed to parse backend")
-	}
-	if tls {
-		if u.Scheme != "" {
-			return fmt.Errorf("scheme expected to be empty")
+// Parse accepts http://ip:port, http://ip or ip:port
+// returns error if backend is invalid and returns hostname if it is valid
+// parses backend string if it is a valid string based on the tlsPassthrough parameter
+// ip:port is only valid in case of tlsPassthrough is true
+// http://ip:port or http://ip is valid in case of tlsPassthrough is false
+func (b Backend) Parse(tlsPassthrough bool) (string, error) {
+	var hostName string
+	if tlsPassthrough {
+		host, port, err := net.SplitHostPort(string(b))
+		if err != nil {
+			return "", fmt.Errorf("failed to parse backend %s with error: %w", b, err)
 		}
-		if u.Port() == "" {
-			return fmt.Errorf("missing port in backend address")
+
+		parsedPort, err := strconv.ParseUint(port, 10, 64)
+		if err != nil {
+			return "", fmt.Errorf("invalid port in backend: %s", port)
 		}
-	} else if u.Scheme != "http" {
-		return fmt.Errorf("scheme expected to be http")
+
+		if parsedPort > math.MaxUint16 {
+			return "", fmt.Errorf("port '%s' must be <= 65535", port)
+		}
+
+		hostName = host
```

Comment:
this is wrong because you need to return `ip:port` in case of tls, or `http://ip:port` in case of no tls